pyuvdata/telescopes.py

```diff
@@ -71,21 +90,163 @@
 }
 
 
+def known_telescopes():
+    """
+    Get list of known telescopes.
+
+    Returns
+    -------
+    list of str
+        List of known telescope names.
+    """
+    astropy_sites = [site for site in EarthLocation.get_site_names() if site != ""]
+    known_telescopes = list(set(astropy_sites + list(KNOWN_TELESCOPES.keys())))
+    return known_telescopes
+
+
+def known_telescope_location(
+    name: str,
+    return_citation: bool = False,
+    known_telescope_dict: dict = KNOWN_TELESCOPES,
+):
+    """
+    Get the location for a known telescope.
+
+    Parameters
+    ----------
+    name : str
+        Name of the telescope
+    return_citation : bool
+        Option to return the citation.
+    known_telescope_dict: dict
+        This should only be used for testing. This allows passing in a
+        different dict to use in place of the KNOWN_TELESCOPES dict.
+
+    Returns
+    -------
+    location : EarthLocation
+        Telescope location as an EarthLocation object.
+    citation : str, optional
+        Citation string.
+
+    """
+    astropy_sites = EarthLocation.get_site_names()
+    telescope_keys = list(known_telescope_dict.keys())
+    telescope_list = [tel.lower() for tel in telescope_keys]
```

Comment:
you should just make a new dict here...

```python
known_telescopes = {k.lower(): v for k, v in known_telescopes_dict}
```

then later you can simply key into the dictionary rather than doing list-index kungfu. Also, this is a good way to ensure that down the function you don't accidentally edit the `KNOWN_TELESCOPES` global dict...